internal/notifier/client.go

```diff
@@ -32,29 +32,90 @@ import (
 	"github.com/hashicorp/go-retryablehttp"
 )
 
-type requestOptFunc func(*retryablehttp.Request)
+type postOption struct {
+	proxy             string
+	certPool          *x509.CertPool
+	requestModifiers  []requestModifier
+	responseValidator func(*http.Response) bool
```

Comment:
Validators should return an `error`, not a boolean. Maybe the response body has information about the error and we need to bubble it up so the error gets logged